http/logical.go

```diff
@@ -178,6 +178,8 @@ func buildLogicalRequestNoAuth(perfStandby bool, w http.ResponseWriter, r *http.
 			path += "/"
 		}
 
+		data = parseQuery(r.URL.Query())
```

Comment:
I don't see why we couldn't do this, but perhaps there is some context I'm missing 🤔 If we don't want to allow query parameters for LIST operations, I'm okay changing this capability to use a GET.

Reply:
Technically the "GET" verb can also translate to a `logical.ListOperation`, but [that case omits](https://github.com/hashicorp/vault/blob/8ae202debcd821ad9b3f54188507ef4500e983c7/http/logical.go#L82) parsing and passing in the rest of the query parameters into the `logical.Request.Data`. 

Should we also consider how to handle query params there if we make it available here? Would there be any backwards compatibility concerns if so?

Reply:
Thanks, that's a good callout. I can't think of any compatibility concerns. The "GET" with `?list=true` would result in a `ListOperation` handler being called, and it would be up to that handler implementation to parse data from query parameters. I don't think any `ListOperation` handlers are parsing parameters today, so I can't think of a concern.

That said, I'm okay with just allowing query parameters for "LIST" at this point. It's a small change, and it makes sense to use query parameters to filter lists. If users want consistent behavior with the GET-style list, we can always address it.

Reply:
I can see the value in keeping the behavior of GET with ?list=true and LIST the same in regards to query params. But I agree that it is a small change and we could wait to see if users want this available for GET ?list=true.